backend/src/models/User.js

```diff
@@ -40,15 +41,40 @@ const userSchema = new Schema({
   },
 });
 
-userSchema.pre('save', function (next) {
+userSchema.pre('save', async function (next) {
   if (!this.isModified('password')) {
     return next();
   }
   // TODO: hash the password here with bcryptjs ref. https://stackoverflow.com/a/53431995
-  this.password = 'hashedPassword';
-  return next();
+  try {
+    this.password = await bcrypt.hash(this.password, 10);
+    return next();
+  } catch (error) {
+    next(error);
+  }
 });
 
+userSchema.methods.validatePassword = async function (password) {
+  return await bcrypt.compare(password, this.password);
+};
+
+
+
+userSchema.path('email').validate(function (value) {
+  const emailregx =
+    /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
This regex is fine for now.
But, moving forward, we shall use [the validator](https://www.npmjs.com/package/validator) package to validate strings.